File: lib/day1.rb
module Fuel
  extend self

  def for_mass_simple(m)
    ( m / 3 ).floor - 2
  end

  def for_mass(m)
    total = 0
    loop do
      moar_fuel = for_mass_simple(m)
      break if moar_fuel < 1
      total += moar_fuel
      m = moar_fuel
    end

    total
  end
end

File: day1.py
from math import floor

def for_mass_simple(m):
    return floor( m / 3 ) - 2

def for_mass(m):
    total = 0
    while True:
        moar_fuel = for_mass_simple(m)
        if moar_fuel < 1:
            break
        total += moar_fuel
        m = moar_fuel

    return total
